src/components/Disturber.tsx

```diff
@@ -0,0 +1,186 @@
+export const Disturber = ({ navigation, staticContentName }: Props) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const refreshTimeKey = `publicJsonFile-${staticContentName}`;
+
+  const { data, refetch } = useStaticContent<DataItem[]>({
+    refreshTimeKey,
+    name: staticContentName,
+    type: 'json'
+  });
+
+  useHomeRefresh(refetch);
+
+  // find the closest item to the current date/time
+  const closestItem: DataItem | null = findClosestItem(
+    data?.filter((item) => item && isActive(item)) || []
+  );
+
+  const setDisturberComplete = () => {
+    setIsVisible((prev) => !prev);
+    addToStore(staticContentName, closestItem?.id.toString());
+  };
+
+  useEffect(() => {
+    const disturberStatus = async () => {
+      try {
+        const disturberComplete = await readFromStore(staticContentName);
+
+        if (closestItem?.id && disturberComplete !== closestItem?.id.toString()) {
+          setIsVisible(true);
+        }
+      } catch (e) {
+        setIsVisible(false);
+
+        console.error(e);
+      }
+    };
+
+    disturberStatus();
+  }, [closestItem]);
+
+  if (!isVisible || !closestItem) return null;
+
+  return (
+    <Wrapper style={styles.wrapperPadding}>
+      <View style={[styles.containerRadius, { backgroundColor: closestItem.backgroundColor }]}>
+        {!!closestItem && (
+          <>
+            <TouchableOpacity onPress={setDisturberComplete} style={styles.closeButton}>
+              <Icon.Close color={colors.darkText} size={normalize(16)} />
+            </TouchableOpacity>
+
+            {!!closestItem.picture && (
+              <Image
+                source={closestItem.picture}
+                borderRadius={normalize(8)}
+                aspectRatio={closestItem.picture.aspectRatio || { HEIGHT: 0.7, WIDTH: 1 }}
+                resizeMode="cover"
+              />
+            )}
+
+            <Wrapper style={styles.noPaddingBottom}>
+              <WrapperHorizontal>
+                {!!closestItem.headline && (
+                  <BoldText center uppercase style={styles.headlineText}>
+                    {closestItem.headline}
+                  </BoldText>
+                )}
+              </WrapperHorizontal>
+            </Wrapper>
+
+            <Wrapper style={styles.noPaddingTop}>
+              <WrapperHorizontal>
+                {!!closestItem.title && (
+                  <BoldText center large>
+                    {closestItem.title}
+                  </BoldText>
+                )}
+              </WrapperHorizontal>
+            </Wrapper>
+
+            <Wrapper style={styles.noPaddingTop}>
+              <WrapperHorizontal>
+                {!!closestItem.description && (
+                  <RegularText center>{closestItem.description}</RegularText>
+                )}
+              </WrapperHorizontal>
+            </Wrapper>
+
+            <Wrapper>
+              <WrapperHorizontal>
+                {!!closestItem.button && closestItem.button.title && (
```

Comment:
as the title is a string, please add `!!` to boolean it

Reply:
Yes, you are right. I fixed it.

https://github.com/smart-village-solutions/smart-village-app-app/pull/909/commits/790533a892a73135a641652cecc7dd521444636d